crates/nu-command/src/conversions/into/datetime.rs

```diff
@@ -304,7 +304,35 @@ fn action(input: &Value, args: &Arguments, head: Span) -> Value {
                                 return Value::date(dt_fixed, span);
                             }
                             ParseResult::DateTime(date) => {
-                                return Value::date(date.fixed_offset(), span)
+                                let local_offset = *Local::now().offset();
+                                let datetime = match local_offset.from_local_datetime(&date) {
+                                    chrono::LocalResult::Single(dt) => dt,
+                                    chrono::LocalResult::Ambiguous(_, _) => {
+                                        return Value::error(
+                                            ShellError::DatetimeParseError {
+                                                msg: "Ambiguous datetime".to_string(),
+                                                span,
+                                            },
+                                            span,
+                                        );
+                                    }
+                                    chrono::LocalResult::None => {
+                                        return Value::error(
+                                            ShellError::DatetimeParseError {
+                                                msg: "Invalid datetime".to_string(),
+                                                span,
+                                            },
+                                            span,
+                                        );
+                                    }
+                                };
+                                let dt_fixed = TimeZone::from_local_datetime(
+                                    &local_offset,
+                                    &datetime.naive_local(),
+                                )
+                                .single()
+                                .unwrap_or_default();
+                                return Value::date(dt_fixed, span);
```

Comment:
could you explain what exactly this is doing, was this prompted by a change in human-date-parser 3.0?